crates/sui/tests/reconfiguration_tests.rs

```diff
@@ -521,86 +522,434 @@ async fn test_reconfig_with_committee_change_basic() {
-            CallArg::Pure(bcs::to_bytes(new_validator.narwhal_primary_address()).unwrap()),
-            CallArg::Pure(bcs::to_bytes(new_validator.narwhal_worker_address()).unwrap()),
+            CallArg::Pure(bcs::to_bytes(&val.network_address()).unwrap()),
+            CallArg::Pure(bcs::to_bytes(&val.p2p_address()).unwrap()),
+            CallArg::Pure(bcs::to_bytes(&val.narwhal_primary_address()).unwrap()),
+            CallArg::Pure(bcs::to_bytes(&val.narwhal_worker_address()).unwrap()),
             CallArg::Pure(bcs::to_bytes(&1u64).unwrap()), // gas_price
             CallArg::Pure(bcs::to_bytes(&0u64).unwrap()), // commission_rate
         ],
         10000,
     )
     .unwrap();
-    let transaction = to_sender_signed_transaction(candidate_tx_data, &new_validator_key);
-    let effects = execute_transaction(&authorities, transaction)
-        .await
-        .unwrap();
-    assert!(effects.status().is_ok());
+    let transaction =
+        to_sender_signed_transaction(candidate_tx_data, node_config.account_key_pair());
+    let effects = execute_transaction(authorities, transaction).await.unwrap();
+    // assert!(effects.status().is_ok());
+    if let ExecutionStatus::Failure {
+        error: e,
+        command: c,
+    } = effects.status().clone()
+    {
+        panic!("Add validator candidate tx unsuccessful: {:?}, {:?}", e, c);
+    } else {
+        info!("Add validator candidate tx successful");
+    }
+    effects_ret.push(effects);
 
     // Step 2: Give the candidate enough stake.
     let stake_tx_data = TransactionData::new_move_call_with_dummy_gas_price(
-        new_validator_address,
+        sender,
         SUI_FRAMEWORK_OBJECT_ID,
         ident_str!("sui_system").to_owned(),
         ident_str!("request_add_stake").to_owned(),
         vec![],
-        gas_objects[1].compute_object_reference(),
+        gas_objects[1],
         vec![
             CallArg::Object(ObjectArg::SharedObject {
                 id: SUI_SYSTEM_STATE_OBJECT_ID,
                 initial_shared_version: SUI_SYSTEM_STATE_OBJECT_SHARED_VERSION,
                 mutable: true,
             }),
-            CallArg::Object(ObjectArg::ImmOrOwnedObject(
-                stake.compute_object_reference(),
-            )),
-            CallArg::Pure(bcs::to_bytes(&new_validator_address).unwrap()),
+            CallArg::Object(ObjectArg::ImmOrOwnedObject(stake)),
+            CallArg::Pure(bcs::to_bytes(&sender).unwrap()),
         ],
         10000,
     )
     .unwrap();
-    let transaction =
-        to_sender_signed_transaction(stake_tx_data, new_node_config.account_key_pair());
-    let effects = execute_transaction(&authorities, transaction)
-        .await
-        .unwrap();
-    assert!(effects.status().is_ok());
+    let transaction = to_sender_signed_transaction(stake_tx_data, node_config.account_key_pair());
+    let effects = execute_transaction(authorities, transaction).await.unwrap();
+    // assert!(effects.status().is_ok());
```

Comment:
Why not just the assert?

Reply:
I actually just changed this back. I was printing the detailed info for debugging purposes. It's good now